### ISSUE TITLE
Configs types are not aligned with source code

### ISSUE THREAD

**Issue by ntnyq** [vuejs/eslint-plugin-vue#2699]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** latest
- **eslint-plugin-vue version:** latest
- **Vue version:** latest
- **Node version:** lts-latest
- **Operating System:** macOS

**Please show your full configuration:**

```json5

```

**What did you do?**	

```vue

```

**What did you expect to happen?**

Types are aligned with source.

**What actually happened?**

Types are not aligned with source.

**Repository to reproduce this issue**

None.

## Types

https://github.com/vuejs/eslint-plugin-vue/blob/d38514020768cb1ea21f577f164cc3b82bcf78c9/lib/index.d.ts#L12-L14

## Source

https://github.com/vuejs/eslint-plugin-vue/blob/d38514020768cb1ea21f577f164cc3b82bcf78c9/lib/index.js#L18-L20

Prefix `vue3-` should be removed.
